src/types/timestamp.rs

```diff
@@ -307,23 +418,18 @@ impl PartialEq for Timestamp {
         {
             return false;
         }
-        if self.precision >= Precision::Second && self_dt.second() != other_dt.second() {
-            return false;
+        if self.precision <= Precision::HourAndMinute {
+            return true;
         }
-        if self.precision >= Precision::FractionalSeconds && !self.fractional_seconds_equal(other) {
+
+        if self_dt.second() != other_dt.second() || !self.fractional_seconds_equal(other) {
             return false;
         }
+
         true
     }
 }
 
-// We cannot provide an implementation of [Ord] for [Timestamp] because many instances cannot be
-// meaningfully compared to each other to derive an ordering. For example, a Timestamp with
-// Precision::Year cannot be compared to a timestamp with Precision::Day, and a Timestamp with
-// an unknown offset cannot be compared to one with a known offset.
-//TODO: It would be possible to provide a PartialOrd for Timestamps with the same precision and
-//      either known or unknown offsets. It's unclear how valuable that would be.
-
```

Comment:
🗺️ This comment is outdated/incorrect. It's possible to provide friendly definitions of `PartialEq` and `PartialOrd` and relegate the precision-aware definition of equality to `IonEq`. See #381.